### PR TITLE
[TESTING] added the memory resource threshold as a config parameter

### PR DESCRIPTION
**Pull request by williamBlazing**

### CODE REVIEW

**Comment by kharoc** [comment]
1 RAL

========================================================
FAILED TESTS                                                                                                                                                                                                
========================================================                                                                                                                                                    
                        TestGroup InputType Result                                           TestId
0  Bindable Alias                  orc       Fail   TEST_09                                        
1  Date                            orc       Fail   TEST_01                                        
2  Simple Distribution From Local  orc       Fail   TEST_22,TEST_43,TEST_46,TEST_49                
3  TPCH Queries                    orc       Fail   TEST_03,TEST_05,TEST-08,TEST_10,TEST_14,TEST_18
4  Timestamp                       orc       Fail   TEST_02                                        
*******************************************************************************
TOTAL SUMMARY for test suite: 
PASSED: 1261/1274
FAILED: 13/1274
CRASH: 0/1274
TOTAL: 1274
MAX DELTA: 156.0
*******************************************************************************
Aggregations without group by:   Start Mem: 427.875   End Mem: 543.875   Diff: 116.0
Coalesce:   Start Mem: 543.875   End Mem: 611.875   Diff: 68.0
Column Basis:   Start Mem: 611.875   End Mem: 611.875   Diff: 0.0
Common Table Expressions:   Start Mem: 611.875   End Mem: 583.875   Diff: -28.0
Count without group by:   Start Mem: 583.875   End Mem: 593.875   Diff: 10.0
Date:   Start Mem: 593.875   End Mem: 655.875   Diff: 62.0
Timestamp:   Start Mem: 655.875   End Mem: 593.875   Diff: -62.0
Full outer join:   Start Mem: 593.875   End Mem: 567.875   Diff: -26.0
Group by:   Start Mem: 567.875   End Mem: 669.875   Diff: 102.0
Group by without aggregations:   Start Mem: 669.875   End Mem: 661.875   Diff: -8.0
Inner join:   Start Mem: 661.875   End Mem: 699.875   Diff: 38.0
Left outer join:   Start Mem: 699.875   End Mem: 677.875   Diff: -22.0
Non-EquiJoin Queries:   Start Mem: 677.875   End Mem: 809.875   Diff: 132.0
Nested Queries:   Start Mem: 809.875   End Mem: 745.875   Diff: -64.0
Order by:   Start Mem: 745.875   End Mem: 851.875   Diff: 106.0
Predicates With Nulls:   Start Mem: 851.875   End Mem: 807.875   Diff: -44.0
Simple String:   Start Mem: 807.875   End Mem: 867.875   Diff: 60.0
Unary ops:   Start Mem: 925.875   End Mem: 1023.875   Diff: 98.0
Unify Tables:   Start Mem: 1023.875   End Mem: 977.875   Diff: -46.0
Union:   Start Mem: 977.875   End Mem: 1039.875   Diff: 62.0
Limit:   Start Mem: 1039.875   End Mem: 961.875   Diff: -78.0
Where clause:   Start Mem: 961.875   End Mem: 1079.875   Diff: 118.0
Bindable Alias:   Start Mem: 1079.875   End Mem: 1081.875   Diff: 2.0
Boolean:   Start Mem: 1081.875   End Mem: 1147.875   Diff: 66.0
Case:   Start Mem: 1147.875   End Mem: 1169.875   Diff: 22.0
Cast:   Start Mem: 1169.875   End Mem: 1247.875   Diff: 78.0
Concat:   Start Mem: 1247.875   End Mem: 1291.875   Diff: 44.0
Literal:   Start Mem: 1291.875   End Mem: 1401.875   Diff: 110.0
Dir:   Start Mem: 1401.875   End Mem: 1499.875   Diff: 98.0
File System Local:   Start Mem: 1499.875   End Mem: 1409.875   Diff: -90.0
File System S3:   Start Mem: 1409.875   End Mem: 1455.875   Diff: 46.0
File System Google Storage :   Start Mem: 1455.875   End Mem: 1429.875   Diff: -26.0
Like:   Start Mem: 1429.875   End Mem: 1527.875   Diff: 98.0
Substring:   Start Mem: 1683.875   End Mem: 1679.875   Diff: -4.0
Wild Card:   Start Mem: 1679.875   End Mem: 1835.875   Diff: 156.0
TPCH Queries:   Start Mem: 1835.875   End Mem: 1929.875   Diff: 94.0
Round:   Start Mem: 1929.875   End Mem: 2003.875   Diff: 74.0

2 Worker's

========================================================
FAILED TESTS                                                                                                                                                                                                
========================================================                                                                                                                                                    
                        TestGroup InputType Result                                           TestId
0  Bindable Alias                  orc       Fail   TEST_09                                        
1  Date                            orc       Fail   TEST_01                                        
2  Simple Distribution From Local  orc       Fail   TEST_22,TEST_43,TEST_46,TEST_49                
3  TPCH Queries                    orc       Fail   TEST_03,TEST_05,TEST-08,TEST_10,TEST_14,TEST_18
4  Timestamp                       orc       Fail   TEST_02                                        
*******************************************************************************
TOTAL SUMMARY for test suite: 
PASSED: 1500/1513
FAILED: 13/1513
CRASH: 0/1513
TOTAL: 1513
MAX DELTA: 475.0
*******************************************************************************
Aggregations without group by:   Start Mem: 844.875   End Mem: 1319.875   Diff: 475.0
Coalesce:   Start Mem: 1319.875   End Mem: 1411.875   Diff: 92.0
Common Table Expressions:   Start Mem: 1411.875   End Mem: 1429.875   Diff: 18.0
Count without group by:   Start Mem: 1429.875   End Mem: 1475.875   Diff: 46.0
Date:   Start Mem: 1475.875   End Mem: 1493.875   Diff: 18.0
Timestamp:   Start Mem: 1493.875   End Mem: 1499.875   Diff: 6.0
Full outer join:   Start Mem: 1499.875   End Mem: 1491.875   Diff: -8.0
Group by:   Start Mem: 1491.875   End Mem: 1531.875   Diff: 40.0
Group by without aggregations:   Start Mem: 1531.875   End Mem: 1615.875   Diff: 84.0
Inner join:   Start Mem: 1615.875   End Mem: 1563.875   Diff: -52.0
Left outer join:   Start Mem: 1563.875   End Mem: 1653.875   Diff: 90.0
Non-EquiJoin Queries:   Start Mem: 1653.875   End Mem: 1803.875   Diff: 150.0
Nested Queries:   Start Mem: 1803.875   End Mem: 1915.875   Diff: 112.0
Order by:   Start Mem: 1915.875   End Mem: 1901.875   Diff: -14.0
Predicates With Nulls:   Start Mem: 1901.875   End Mem: 2055.875   Diff: 154.0
Simple String:   Start Mem: 2055.875   End Mem: 2061.875   Diff: 6.0
Unary ops:   Start Mem: 2253.875   End Mem: 2191.875   Diff: -62.0
Unify Tables:   Start Mem: 2191.875   End Mem: 2283.875   Diff: 92.0
Union:   Start Mem: 2283.875   End Mem: 2135.875   Diff: -148.0
Limit:   Start Mem: 2135.875   End Mem: 2077.875   Diff: -58.0
Where clause:   Start Mem: 2077.875   End Mem: 2111.875   Diff: 34.0
Bindable Alias:   Start Mem: 2111.875   End Mem: 2149.875   Diff: 38.0
Boolean:   Start Mem: 2149.875   End Mem: 2219.875   Diff: 70.0
Case:   Start Mem: 2219.875   End Mem: 2267.875   Diff: 48.0
Cast:   Start Mem: 2267.875   End Mem: 2315.875   Diff: 48.0
Concat:   Start Mem: 2315.875   End Mem: 2419.875   Diff: 104.0
Literal:   Start Mem: 2419.875   End Mem: 2477.875   Diff: 58.0
Dir:   Start Mem: 2477.875   End Mem: 2839.875   Diff: 362.0
File System Local:   Start Mem: 2839.875   End Mem: 2705.875   Diff: -134.0
File System S3:   Start Mem: 2705.875   End Mem: 2779.875   Diff: 74.0
File System Google Storage :   Start Mem: 2779.875   End Mem: 2749.875   Diff: -30.0
Like:   Start Mem: 2749.875   End Mem: 3045.875   Diff: 296.0
Substring:   Start Mem: 3235.875   End Mem: 3307.875   Diff: 72.0
Wild Card:   Start Mem: 3307.875   End Mem: 3447.875   Diff: 140.0
TPCH Queries:   Start Mem: 3447.875   End Mem: 3683.875   Diff: 236.0
Round:   Start Mem: 3683.875   End Mem: 3779.875   Diff: 96.0

File: engine/src/bmr/BlazingMemoryResource.h [hunk 1]
```diff
@@ -171,7 +171,7 @@ class blazing_device_memory_resource : public BlazingMemoryResource {
    * 
    * @param[in] options Optional options to set
    * ----------------------------------------------------------------------**/
-    void initialize(const rmmOptions_t *new_options) {
+    void initialize(const rmmOptions_t *new_options, float BLAZING_DEVICE_MEM_RESOURCE_CONSUMPTION_THRESHOLD) {
```
Comment (felipeblazing) [inline engine/src/bmr/BlazingMemoryResource.h:174]: does the fact that this is some config merit an all caps deviation from normal parameter naming?

File: engine/src/cython/initialize.cpp [hunk 2]
```diff
@@ -151,5 +153,11 @@ void blazingSetAllocator(
 	for (size_t i = 0; i < devices.size(); ++i)
 		rmmValues.devices.push_back(devices[i]);
 
-	BlazingRMMInitialize(&rmmValues);
+	float BLAZING_DEVICE_MEM_RESOURCE_CONSUMPTION_THRESHOLD = 0.95;
```
Comment (felipeblazing) [inline engine/src/cython/initialize.cpp:156]: use normal naming convention for variables.

File: engine/src/bmr/BlazingMemoryResource.h (hunk 1 above)
Comment (williamBlazing) [reply engine/src/bmr/BlazingMemoryResource.h:174]: Constants are traditionally set as ALL CAPS. We have used that pattern for all configurable parameters in the past. This is sticking with our current naming convention for config parameters which as has been to set them as all caps.

File: engine/src/cython/initialize.cpp (hunk 2 above)
Comment (williamBlazing) [reply engine/src/cython/initialize.cpp:156]: Constants are traditionally set as ALL CAPS. We have used that pattern for all configurable parameters in the past. This is sticking with our current naming convention for config parameters which as has been to set them as all caps.

File: engine/src/bmr/BlazingMemoryResource.h (hunk 1 above)
Comment (felipeblazing) [reply engine/src/bmr/BlazingMemoryResource.h:174]: This isnt a compile time constant. It is actually part of the state of the context.

**Pull request merged**